src/pages/index.js

```diff
@@ -6,173 +7,240 @@ import PopupWithForm from "../components/PopupWithForm.js";
 
-profileEditCloseButton.addEventListener("click", () => {
-  profileEditPopup.close();
-});
-
 // Profile Edit Modal
-function openProfileEditForm() {
+
+const profileEditPopup = new PopupWithForm(
+  "#edit-modal",
+  handleProfileEditFormSubmit
+);
+
+profileEditPopup.setEventListeners();
+
+function fillProfileForm() {
   const profileInfo = userInfo.getUserInfo();
   profileTitleInput.value = profileInfo.name;
   profileDescriptionInput.value = profileInfo.job;
+}
+
+profileEditButton.addEventListener("click", () => {
+  fillProfileForm();
   profileEditPopup.open();
+});
+
+function handleProfileEditFormSubmit(profileInfo) {
+  profileEditPopup.setLoading(true);
+  api
+    .updateUserInfo(profileInfo)
+    .then((res) => {
+      userInfo.setUserInfo(
+        {
+          name: res.name,
+          job: res.about,
+        },
+        profileEditPopup.close()
+      );
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => profileEditPopup.setLoading(false));
 }
 
-profileEditButton.addEventListener("click", openProfileEditForm);
+// Change Avatar Modal
+
+const changeAvatarModal = document.querySelector("#change-avatar-modal");
+const changeImageForm = new PopupWithForm(
+  "#change-avatar-modal",
+  handleChangeAvatarFormSubmit
+);
+changeImageForm.setEventListeners();
+
+const changeAvatarButton = document.querySelector(
+  ".change-avatar__edit-button"
+);
 
-function handleProfileEditFormSubmit(inputValues) {
-  const profileInfo = {};
-  profileInfo.name = inputValues.title;
-  profileInfo.job = inputValues.description;
-  console.log(profileInfo);
-  userInfo.setUserInfo(profileInfo);
-  profileEditPopup.close();
+changeAvatarButton.addEventListener("click", () => {
+  changeImageForm.open();
+});
+
+function handleChangeAvatarFormSubmit(profileInfo) {
+  changeImageForm.setLoading(true);
+  api
+    .updateAvatar(profileInfo)
+    .then((res) => {
+      userInfo.setAvatar({ avatar: res.avatar });
+      changeImageForm.close();
+    })
+    .catch((err) => console.error(err))
+    .finally(() => changeImageForm.setLoading(false));
 }
 
-// Initial Cards
-function handleImageClick(cardData) {
-  popupImage.open(cardData);
+// Delete Card
+
+const deleteConfirmation = new PopupDeleteConfirm(
+  "#delete-image-modal",
+  handleDeleteCard
+);
+deleteConfirmation.setEventListeners();
+
+function handleDeleteCard(cardData) {
+  deleteConfirmation.open();
+  deleteConfirmation.confirmDelete(() => {
+    deleteConfirmation.setLoading(true);
+    api
+      .deleteCard(cardData.getCardId())
+      .then(() => {
+        cardData.removeCard();
+        deleteConfirmation.close();
+      })
+      .catch((err) => console.error(err))
+      .finally(() => deleteConfirmation.setLoading(false));
+  });
 }
 
+//Liking and Disliking Cards
+
+function handleCardLike(cardData) {
+  if (!cardData.like) {
+    api.likeCard(cardData.getCardId()).then(() => {
+      cardData.handleLike(true);
+    });
```

Comment:
You need to `catch` possible errors at the very **end** of any server request.

**Please, correct it everywhere**